request-server.go

```diff
@@ -107,11 +107,11 @@ func (rs *RequestServer) Serve() error {
 	var pktType uint8
 	var pktBytes []byte
 	for {
-		pktType, pktBytes, err = rs.recvPacket()
+		pktType, pktBytes, err = rs.recvPacket(rs.pktMgr.alloc, rs.pktMgr.getNextOrderID())
```

Comment:
Here we’re calling a receiver method on `rs` and passing values from the `rs` object as arguments. We shouldn’t do that, the receiver method will already have access to these values.

Reply:
You should read this as `rs.serverConn.recvPacket(rs.pktMgr.alloc, rs.pktMgr.getNextOrderID())`

anyway I'll add alloc to conn struct as you suggested before, so this could be simplified in:

```
pktType, pktBytes, err = rs.recvPacket(rs.pktMgr.getNextOrderID())
```